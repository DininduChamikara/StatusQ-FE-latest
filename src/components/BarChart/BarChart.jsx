import { Box } from "@mui/material";
import React from "react";
import { Bar } from "react-chartjs-2";

// Do not remove this => import Chart from 'chart.js/auto'; 
// it will gives error "category" is not a registered scale
// eslint-disable-next-line no-unused-vars
import Chart from 'chart.js/auto';
// import Chart from 'chart.js/auto';

function BarChart({barChartData}) {

  const data = {
    // labels: ["O/L", "A/L", "Undergraduates", "Postgraduates"],
    labels: barChartData.labelsArray,
    datasets: [
      {
        label: "Audience Count",
        data: barChartData.dataArray,
        backgroundColor: [
          "rgba(255, 99, 132, 0.2)",
          "rgba(255, 159, 64, 0.2)",
          "rgba(255, 205, 86, 0.2)",
          "rgba(75, 192, 192, 0.2)",
          "rgba(54, 162, 235, 0.2)",
          "rgba(153, 102, 255, 0.2)",
          "rgba(201, 203, 207, 0.2)",
        ],
        borderColor: [
          "rgb(255, 99, 132)",
          "rgb(255, 159, 64)",
          "rgb(255, 205, 86)",
          "rgb(75, 192, 192)",
          "rgb(54, 162, 235)",
          "rgb(153, 102, 255)",
          "rgb(201, 203, 207)",
        ],
        borderWidth: 1,
      },
    ],
  };

  const chartOptions = {
    scales: {
      xAxes: [
        {
          stacked: true,
          barPercentage: 0.2,
        },
      ],
      yAxes: [
        {
          stacked: true,
          barPercentage: 0.2,
        },
      ],
    },
  };

  return (
    <Box data-testid="barChart">
      <Bar data={data} options={chartOptions} />
    </Box>
  );
}

export default BarChart;
