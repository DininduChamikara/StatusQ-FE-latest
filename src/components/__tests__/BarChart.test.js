import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import BarChart from "../BarChart/BarChart";

describe("BarChart", () => {
  it("renders the BarChart with the correct data", () => {
    const barChartData = {
      labelsArray: ["O/L", "A/L", "Undergraduates", "Postgraduates"],
      dataArray: [10, 20, 30, 40],
    };

    const { getByTestId } = render(<BarChart barChartData={barChartData} />);

    // eslint-disable-next-line testing-library/prefer-screen-queries
    const chartElement = getByTestId("barChart");
    expect(chartElement).toBeInTheDocument();
  });

  it("BarChart matches snapshot", () => {
    const barChartData = {
      labelsArray: ["O/L", "A/L", "Undergraduates", "Postgraduates"],
      dataArray: [10, 20, 30, 40],
    };

    const tree = renderer
      .create(<BarChart barChartData={barChartData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
