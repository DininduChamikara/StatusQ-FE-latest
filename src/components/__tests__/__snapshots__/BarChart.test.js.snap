// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarChart BarChart matches snapshot 1`] = `
<div
  className="MuiBox-root css-0"
  data-testid="barChart"
>
  <canvas
    height={150}
    role="img"
    width={300}
  />
</div>
`;
